syntax = "proto3";

package glgui;

// Rectangle message with x, y coordinates for the top-left and bottom-right corners
message RectangleResponse {
  float x1 = 1;
  float y1 = 2;
  float x2 = 3;
  float y2 = 4;
}

// Empty message for request
message RectangleRequest {}

message GetImageRequest {
  int32 width = 1;
  int32 height = 2;
}

message GetImageResponse {
  int32 width = 1;
  int32 height = 2;
  bytes data = 3;
}



service GLGuiService {
  // RPC method to get a random rectangle
  rpc GetRandomRectangle (RectangleRequest) returns (RectangleResponse);
  rpc GetImage (GetImageRequest) returns (GetImageResponse);
}